### PR TITLE
fix: fire Dragover events to accompany DragEnter and Drop

### LINKED ISSUE #15955: [Skia] TreeView drag glyphs issue
### Current behavior

The glyph shows a block between items on Skia heads

6wDEvy5KRg

### Expected behavior

vDc5hjQC9h

### How to reproduce it (as minimally and precisely as possible)

[TreeViewRepro.zip](https://github.com/unoplatform/uno/files/14665585/TreeViewRepro.zip)

### Workaround

_No response_

### Works on UWP/WinUI

Yes

### Environment

_No response_

### NuGet package version(s)

5.1.87

### Affected platforms

Skia (WPF), Skia (GTK on Linux/macOS/Windows), Skia (Linux Framebuffer)

### IDE

Visual Studio 2022

### IDE version

_No response_

### Relevant plugins

_No response_

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by ramezgerges**
GitHub Issue (If applicable): closes #15955

## PR Type

What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/UIElement.Pointers.cs [hunk 1]
```diff
@@ -792,13 +792,18 @@ private protected virtual void PrepareShare(DataPackage data)
...
+			var wasDragOver = IsDragOver(args.SourceId);
 
 			(_draggingOver ??= new HashSet<long>()).Add(args.SourceId);
 
-			SafeRaiseEvent(evt, args);
+			if (!wasDragOver)
+			{
+				var args2 = new DragEventArgs(args);
+				SafeRaiseEvent(DragEnterEvent, args);
+				args.AcceptedOperation = args2.AcceptedOperation;
```
Comment (dr1rrb) [inline src/Uno.UI/UI/Xaml/UIElement.Pointers.cs:803]: Why creating a new `DragEventArgs` just to "save" the `AcceptedOperation` (you are not using it in the `SafeRaiseEvent` ?

I think this is enough:
```suggestion
				SafeRaiseEvent(DragEnterEvent, args);
				args.Handled = false;
```

File: src/Uno.UI/UI/Xaml/UIElement.Pointers.cs [hunk 2]
```diff
@@ -813,7 +818,14 @@ internal void RaiseDrop(global::Microsoft.UI.Xaml.DragEventArgs args)
...
 			{
-				SafeRaiseEvent(DropEvent, args);
+				var args2 = new DragEventArgs(args);
+				SafeRaiseEvent(DragOverEvent, args2);
+				args.AcceptedOperation = args2.AcceptedOperation;
+				// On WinUI, if the final DragOver event sets AcceptedOperation to None, the DropEvent won't fire
+				if (args.AcceptedOperation is not DataPackageOperation.None)
+				{
+					SafeRaiseEvent(DropEvent, args);
+				}
```
Comment (dr1rrb) [inline src/Uno.UI/UI/Xaml/UIElement.Pointers.cs:828]: Same here:
```suggestion
				SafeRaiseEvent(DragOverEvent, args);
				
				// On WinUI, if the final DragOver event sets AcceptedOperation to None, the DropEvent won't fire
				if (args.AcceptedOperation is not DataPackageOperation.None)
				{
					args.Handled = false;
					SafeRaiseEvent(DropEvent, args);
				}
```

Comment (dr1rrb) [reply src/Uno.UI/UI/Xaml/UIElement.Pointers.cs:828]: Also _"On WinUI, if the final DragOver event sets AcceptedOperation to None, the DropEvent won't fire"_ : Is there any internal state that is clean in drop event handlers? If so we should make sure to clear it anyway.

**Review by dr1rrb — APPROVED** [review]

File: src/Uno.UI/UI/Xaml/UIElement.Pointers.cs (hunk 1 above)
Comment (ramezgerges) [reply src/Uno.UI/UI/Xaml/UIElement.Pointers.cs:803]: That's how WinUI does it. The 2 args are not the same. I do think it's unnecessary too (your suggestion was the initial implementation), so I'm on board if it's okay to not match WinUI in this instance.

File: src/Uno.UI/UI/Xaml/UIElement.Pointers.cs (hunk 2 above)
Comment (ramezgerges) [reply src/Uno.UI/UI/Xaml/UIElement.Pointers.cs]: If I understand correctly, UIElementss should not expect to get a `DragDrop` every time. Cleaning up should instead happen in `DragCompleted` (which has a `Canceled` property), which we apparently don't actually raise.

**Review by dr1rrb — APPROVED** [review]

**Pull request merged**